### PR TITLE
Revamp the initial support for driving on the left

### PR DESCRIPTION
**Pull request by dabreegster**
My first attempt at handling left-hand driving was to take all of the code that shifts a road center line left/right and just sort of invert it. That actually did work, but it was extremely confusing and was an API change that percolated everywhere. Now that we have `lanes_ltr` and not `children_fwd/back`, rip out the old hack and handle things in just one smaller, unit tested place.

@michaelkirk

### CODE REVIEW

File: game/src/layer/map.rs [hunk 1]
```diff
@@ -219,7 +219,7 @@ impl Static {
         let edits = app.primary.map.get_edits();
         for r in &edits.changed_roads {
             let r = app.primary.map.get_r(*r);
-            let orig = EditRoad::get_orig_from_osm(r);
+            let orig = EditRoad::get_orig_from_osm(r, app.primary.map.get_config().driving_side);
```
Comment (michaelkirk) [inline game/src/layer/map.rs:222]: God help us when we get to mapping the hong-kong / mainland china border.

Comment (michaelkirk) [reply game/src/layer/map.rs:222]: image

Comment (dabreegster) [reply game/src/layer/map.rs:222]: It'll actually be easy, just refactor this into a function that maps a way to left/right! The cars will wiggle around each other correctly

File: map_model/src/make/initial/lane_specs.rs [hunk 2]
```diff
@@ -371,11 +398,26 @@ mod tests {
                     "cycleway:right:oneway=no",
                     "oneway:bicycle=no",
                 ],
+                DrivingSide::Right,
                 "sddbbs",
                 "v^^v^^",
             ),
+            (
```
Comment (michaelkirk) [inline map_model/src/make/initial/lane_specs.rs:405]: 👍

**Review by michaelkirk — APPROVED** [review]
There used to be a comment in the DrivingSide impl

>  // "right" and "left" here are in terms of DrivingSide::Right, what I'm used to reasoning about	
>  // in the USA. They invert appropriately for DrivingSide::Left.

... which was deleted. It seems like now, "shift right" always actually means "right" now, even when it's a left-side drive map. Is that correct?

**Comment by dabreegster** [comment]
> There used to be a comment in the DrivingSide impl
> 
> > // "right" and "left" here are in terms of DrivingSide::Right, what I'm used to reasoning about
> > // in the USA. They invert appropriately for DrivingSide::Left.
> 
> ... which was deleted. It seems like now, "shift right" always actually means "right" now, even when it's a left-side drive map. Is that correct?

Yes, now "left" and "right" just follow https://wiki.openstreetmap.org/wiki/Forward_%26_backward,_left_%26_right, no mysterious inversions. When I get around to filling out docstrings in geom, I'll be clear about those.

Thanks for the review!

**Comment by michaelkirk** [comment]
> Yes, now "left" and "right" just follow https://wiki.openstreetmap.org/wiki/Forward_%26_backward,_left_%26_right, no mysterious inversions. When I get around to filling out docstrings in geom, I'll be clear about those.

That's excellent. Good work!

**Pull request merged**